src/hooks/useRequest.ts

```diff
@@ -1,29 +1,37 @@
 import { useState, useEffect } from 'react';
+import type { GitHubData } from '@/types/types';
 
-export default function useRequest(url: string, method = 'GET'): [any, boolean, any, () => void] {
-    const [data, setData] = useState(null);
-    const [err, setError] = useState(null);
+/**
+ * A hook to perform a GET request to the specified URL.
+ * @param url The URL to fetch.
+ * @returns An array with the first element being the data returned from the fetch,
+ * or null, and the second element being a boolean indicating whether the fetch has completed.
+ */
+const useRequest = (url: string): [GitHubData | null, boolean] => {
+     const [data, setData] = useState<GitHubData | null>(null);
```

Comment:
The naming of this file makes no sense anymore. The useRequest hook was a general hook to fetch data from the api but now by the looks of it it has been modified to just fetch the github data from the api? Or at least something like it. Perhaps a refactor or rename makes more sense?